### DISCUSSION TITLE
It is possible join the i-cloud sync

### DISCUSSION THREAD

**Discussion by sixdjango** [toeverything/AFFiNE#2471] [Ideas]
i want use i-cloud sync my all files

**Comment by Dax911** [comment]
while on mac parts of the file system are automatically synced with icloud. So changing the path of storage to a path that is in the shared scope would do the trick... Im looking at doing this with juicefs as im homebrewing my own shared filesystem.

**Reply by Dax911** [reply]
Correction i dont think it exists in the Electron version the only mention of cloud was in a unit test covering displaying the modal.

**Reply by Dax911** [reply]
i found using juicefs sync works great. I believe the ```rsync –r``` tool will work just as well for you.

**Comment by HeJiachen-PM** [comment]
Hello all.
Syncing are of top priority in our development of Octobase, the database and backend framework of AFFiNE. We are not opening direct integration of i-cloud mainly because:
1. iCloud doesn't notify apps when and how it is syncing. It simply broke the .db file when conflict occurs, namely two client currently writing into one paths on icloud. AFFiNE used CRDTs(literally Conflict-free Replicateable Data Types) to avoid such things, but storing one centric copy on a cloud provider for multiple devices isn't a fit for CRDTs. Note that due to possible async io of iCloud, current writing will appear even when you are not using AFFiNE.
2. The suggested way to deal with centric personal cloud such as icloud, onedrive, dropbox all is to develop their specific provider. We believe instead of adapting all the providers, develop one robust open source AFFiNE cloud first makes much more sense.

**Reply by Dax911** [reply]
Agreed, and I can't wait to see it. If you can avoid conflicts I was just giving them a temporary solution. LMK if there is an issue I can tackle to help out.

**Comment by HeJiachen-PM** [comment]
Ofc, AFFiNE cloud will make it easier for self-host with docker, too.

**Comment by sixdjango** [comment]
Thank you all for answer, very much looking forward to AFFiNE cloud! roughly when to beta
